anchor/database/src/lib.rs

```diff
@@ -228,3 +229,78 @@ impl NetworkDatabase {
         });
     }
 }
+
+/// A helper to get the operator ID of the current operator. Caches the ID after successfully
+/// retrieving it to avoid locking the state further.
+pub enum OwnOperatorId {
+    /// The operator ID was known when the `OwnOperatorId` was created.
+    Known(OperatorId),
+    /// The operator ID was not known when the `OwnOperatorId` was created. It will be retrieved
+    /// from the `receiver` and cached in the `id` on first success.
+    FromState {
+        receiver: Receiver<NetworkState>,
+        /// We use a `OnceLock` so that `get` can be called without a mutable reference.
+        id: OnceLock<OperatorId>,
+    },
+}
```

Comment:
Do we need an enum with two variants or could we set the id when the `OperatorId` is available?